setup.py

```diff
@@ -176,7 +176,7 @@ def run(self):
 # - `qjit`: Path to the JIT compiler decorator provided by the compiler.
 setup(
     classifiers=classifiers,
-    name="pennylane-catalyst",
+    name="PennyLane-Catalyst",
```

Comment:
@grwlf my understanding is that for this PR, _only_ this line should be changed, alongside any lines that refer to explicit wheel/egg files. But pip commands should remain as-is

Reply:
Got it, I have reverted the changes other than this line and the egg-related ones